gui/options.cpp

```diff
@@ -342,6 +347,37 @@ void OptionsDialog::build() {
 			_stretchPopUp->setVisible(false);
 		}
 
+		_scalerPopUp->setSelected(0);
+		_scaleFactorPopUp->setSelected(0);
+
+		if (g_system->hasFeature(OSystem::kFeatureScalers)) {
+			if (ConfMan.hasKey("scaler", _domain)) {
+				const PluginList &scalerPlugins = ScalerMan.getPlugins();
+				Common::String scaler(ConfMan.get("scaler", _domain));
+
+				for (uint scalerIndex = 0; scalerIndex < scalerPlugins.size(); scalerIndex++) {
+					if (scumm_stricmp(scalerPlugins[scalerIndex]->get<ScalerPluginObject>().getName(), scaler.c_str()) != 0)
+						continue;
+
+					_scalerPopUp->setSelectedTag(scalerIndex);
+					updateScaleFactors(scalerIndex);
+
+					if (ConfMan.hasKey("scale_factor", _domain)) {
+						int scaleFactor = ConfMan.getInt("scale_factor", _domain);
+						if (scalerPlugins[scalerIndex]->get<ScalerPluginObject>().hasFactor(scaleFactor))
```

Comment:
What will happen if switching to a non-existent scale? Say, from Normal 4x to SAI, which has only 2x?

Reply:
The scale factor drop-down is automatically updated whenever the selected scaler changes, which prevents invalid combinations from being selected in the GUI. If an invalid scale factor is specified via the `scummvm.ini`, the previously used factors from the backend and the individual scaler plugins will be used as fallbacks.